ibis/expr/operations.py

```diff
@@ -2746,6 +2746,27 @@ def output_type(self):
         return rules.shape_like(self.args[0], map_type.value_type)
 
 
+class MapValueOrDefaultForKey(ValueOp):
+
+    input_type = [
+        rules.map(dt.any, dt.any),
+        rules.one_of((dt.string, dt.int_), name='key'),
+        rules.value(name='default')
+    ]
+
+    def output_type(self):
+        map_type = self.args[0].type()
+        value_type = map_type.value_type
+        default_type = self.args[2].type()
```

Comment:
you can do `self.default.type()` here which is a bit more readable.

Reply:
done